app/src/views/DatasetManagementView.js

```diff
@@ -54,12 +29,26 @@
                     dataset = this.datasets.get(this.$('.datasets').val());
                 flow.retrieveDatasetAsFormat(dataset, dataset.get('type'), format, false, _.bind(function (error, converted) {
                     var blob = new Blob([converted.get('data')]),
-                        extension = this.extensions[dataset.get('type') + ":" + format],
+                        extension = _.first(flow.getExtensionsFromTypeFormat(dataset.get('type'), format)),
                         parts = name.split('.'),
                         nameWithExtension = parts[parts.length - 1] === extension ? name : name + '.' + extension,
-                    file = flow.girderUpload(blob, nameWithExtension, flow.saveLocation.get('dataFolder'), false, function () {
-                        dataset.set({id: file.get('itemId')});
-                    });
+                        file = flow.girderUpload(blob, nameWithExtension, flow.saveLocation.get('dataFolder'), false, function () {
+                            // Save type/format on the item for retrievel later
+                            // These get set directly on the dataset model
+                            var item = new girder.models.ItemModel({
+                                _id: file.get('itemId')
+                            }).fetch();
+
+                            // _.noop Can be undefined with girder/girder#1057
+                            item.addMetadata('flow', {
+                                type: dataset.get('type'),
+                                format: dataset.get('format')
+                            }, _.noop, function () {
+                                window.flow.bootstrapAlert("danger", "Failed to set metadata on item");
+                            });
```

Comment:
The file upload and the metadata set should both be treated atomically. Perhaps using something like the facilities mentioned in girder/girder#1072.